src/js/vic-v2/components/PhotoField.jsx

```diff
@@ -231,42 +304,53 @@ export default class PhotoField extends React.Component {
     this.setState({ dragAndDropSupported: dragAndDropSupported && !iOS });
   }
 
-  detectWidth = () => {
-    const windowWidth = window.innerWidth;
-    this.setState({ windowWidth });
-  }
-
-  moveUp = () => {
-    this.refs.cropper.move(0, -5);
-  }
-
-  moveDown = () => {
-    this.refs.cropper.move(0, 5);
-  }
+  updateBoundaryWarningAndButtonStates = () => {
+    const canvasCropBoundariesMet = getCanvasCropBoundariesMet(this.refs.cropper.getCanvasData(), this.refs.cropper.getCropBoxData());
 
-  moveRight = () => {
-    this.refs.cropper.move(-5, 0);
+    this.setState({ ...makeMoveButtonsEnabledStates(canvasCropBoundariesMet) });
+    this.setState({
+      warningMessage: makeCropBoundaryWarningMessage(canvasCropBoundariesMet),
+      ...makeMoveButtonsEnabledStates(canvasCropBoundariesMet)
+    });
   }
 
-  moveLeft = () => {
-    this.refs.cropper.move(5, 0);
+  cropend = () => { // casing matches Cropper argument
+    this.updateBoundaryWarningAndButtonStates();
   }
 
-  zoom = (e) => {
-    if (e.target.value < MAX_RATIO && e.target.value > MIN_RATIO) {
-      this.refs.cropper.zoomTo(e.target.value);
+  handleMove = (direction) => {
```

Comment:
had to remove some functions to get under the code climate warning threshold. zooms and moves are now `handleMove` and `handleZoom`. Also moved all of the pure functions out of the component.